### PR TITLE
[V2] Share functionality in common select

### PR DESCRIPTION
**Pull request by rvsia**
- loadOptions functionality moved to common select
- PF4 select is using common now
- PF3 can load options by searching

### CODE REVIEW

File: packages/common/src/select/index.js [hunk 1]
```diff
@@ -15,29 +20,143 @@ const handleSelectChange = (option, simpleValue, isMulti, onChange) => {
     : onChange(sanitizedOption);
 };
 
-class Select extends Component {
-  render() {
-    const { input, invalid, classNamePrefix, simpleValue, isMulti, pluckSingleValue, options, ...props } = this.props;
-    const { value, onChange, ...inputProps } = input;
+const selectProvider = (type) =>
```
Comment (Hyperkid123) [inline packages/common/src/select/index.js]: I think we should make this just a simple object.

File: packages/common/src/select/index.js [hunk 2]
```diff
@@ -15,29 +20,143 @@ const handleSelectChange = (option, simpleValue, isMulti, onChange) => {
...
+      dispatch({ type: 'updateOptions', payload: data });
+    });
+  };
+
+  useEffect(() => {
+    if (loadOptions) {
+      updateOptions();
+    }
+
+    dispatch({ type: 'mounted' });
```
Comment (Hyperkid123) [inline packages/common/src/select/index.js]: This is not reliable, you should use `useRef` for cases like this. There is an example: https://gist.github.com/jaydenseric/a67cfb1b809b1b789daa17dfe6f83daa

**Comment by Hyperkid123** [comment]
the PF3 shows no data text instead of loading text.

pf3-loading

**Comment by rvsia** [comment]
@Hyperkid123 Everything is fixed, MUI is using the common select too!

**Comment by Hyperkid123** [comment]
@rvsia something hapenned

**Pull request merged**